enhancements/machine-config/mco-cgroupsv2-support.md

```diff
@@ -0,0 +1,206 @@
+not have data around the pure mode environment yet, there needs to be a platform
+to gather data from.
+
+## Proposal
+
+The option to enable cgroup v2 will have to reside in a centralized location.
+The [OpenShift Infrastructure config
+object](https://github.com/openshift/api/blob/master/config/v1/types_infrastructure.go#L28)
+contains information describing how a cluster functions including cloud config  
+and platform specification for each cloud. Setting the cgroup mode is an
+infrastructure setting.
+
+### API Extensions
+
+Create an additional module within openshift/api as `config/v1/types_node.go`:
+
+```go
+type CgroupMode string
+
+const (
+	CgroupMode_v1 = ""
+	CgroupMode_v2 = "v2"
+	CgroupMode_Default = CgroupMode_v1
+)
+
+type NodeOptions struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// +required
+	Spec NodeOptionsSpec `json:"spec"`
+}
+
+type NodeOptionsSpec struct {
+  CgroupMode CgroupMode `json:"cgroupMode,omitempty"`
+
+  // an eventual additional option might be crun in the future. This explains
+  //   why a new struct may be necessary
+  // CrunEnabled bool ...
+}
+```
+
+### Operational Aspects of API Extensions
+
+Once the previous API is defined, the MCO will read the configured object and
+set the appropriate kernel options (on bootstrap). The MCO will report an error
+if a user tries to modify/add cgroup related kargs within a MachineConfig.
+
+
+The following kernel command line arguments would be set when `CgroupMode_v2` is enabled:
+```yaml
+  kernelArguments:
+    - systemd.unified_cgroup_hierarchy=1
+    - cgroup_no_v1="all"
+    - psi=1 
+```
+
+#### Failure Modes
+
+N/A
+
+#### Support Procedures
+
+### User Stories
+
+#### 1. As a user, I would like to install a cluster that uses cgroup v2
+
+### Risks and Mitigations
+
+The primary risk of cgroup v2 is some workloads not supporting the changed
+cgroup filesystem paths. This is the reason why it is off by default and has to
+be enabled at install time.
+
+## Design Details
+
+### Test Plan
+
+Testing should be thoroughly done at all levels, including unit, end-to-end, and
+integration.
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+cgroup v2 will be dev preview on its initial release. Internal and customer
+usage will be critical to gather information on bugs and enhancements to the
+underlying subsystem.
+
+Graduation requirements to Tech Preview are:
+
+* No regressions from cgroup v1 to cgroup v2
+* Processes are correctly OOMed
+* CPU management works as expected
+* No performance issues - PSAP and QE teams will be asked to test their suites for regressions
+* Metrics are accurate and correctly submitted to monitoring
+
+#### Tech Preview -> GA
+
+With sufficient internal testing and customer feedback the feature will graduate
+to Tech Preview.
+
+Upon graduation to GA the feature will still be turned off by default, but may
+be enabled within OpenShift at a later time.
```

Comment:
do you expect a period where it is on by default, but can be disabled?

Reply:
I updated this sentence in f69f9c7 to address that another enhancement will be created to specify how the feature will be turned on by default. The original sentence here was not clear with my other modifications to the document.